skbio/util/_testing.py

```diff
@@ -94,3 +94,12 @@ def get_data_path(fn, subfolder='data'):
     path = os.path.dirname(os.path.abspath(callers_filename))
     data_path = os.path.join(path, subfolder, fn)
     return data_path
+
+
+def _skip_if_no_matplotlib():
+    """Skip rest of a test if matplotlib is not available."""
+    try:
+        import matplotlib  # noqa
+    except ImportError:
+        import nose
+        raise nose.SkipTest("matplotlib not installed")
```

Comment:
There's also Numpy's skipif decorator.
http://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.decorators.skipif.html

On (Apr-01-15| 8:42), Evan Bolyen wrote:
> > @@ -94,3 +94,12 @@ def get_data_path(fn, subfolder='data'):
> >      path = os.path.dirname(os.path.abspath(callers_filename))
> >      data_path = os.path.join(path, subfolder, fn)
> >      return data_path
> > +
> > +
> > +def _skip_if_no_matplotlib():
> > +    """Skip rest of a test if matplotlib is not available."""
> > +    try:
> > +        import matplotlib  # noqa
> > +    except ImportError:
> > +        import nose
> > +        raise nose.SkipTest("matplotlib not installed")
> 
> It might be better to raise `SkipTest` from the [stdlib](https://docs.python.org/2/library/unittest.html#unittest.SkipTest) instead of Nose. Unless there was a specific reason you are using nose's skiptest.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/scikit-bio/pull/886/files#r27581531